sympy/functions/special/tests/test_singularity_functions.py

```diff
@@ -0,0 +1,64 @@
+from sympy import (
+    adjoint, conjugate, nan, pi, symbols, transpose,
+    Piecewise, I, Eq, Derivative, oo, SingularityFunction
+)
+
+
+# from sympy.functions.special.singularity_functions import SingularityFunction
+from sympy.core.function import ArgumentIndexError
+from sympy.utilities.pytest import raises
+
+x, y, a, n = symbols('x y a n')
+
+
+def test_fdiff():
+    assert SingularityFunction(x, 4, 5).fdiff() == 5*SingularityFunction(x, 4, 4)
+    assert SingularityFunction(x, 4, -1).fdiff() == SingularityFunction(x, 4, -2)
+    assert SingularityFunction(x, 4, 0).fdiff() == SingularityFunction(x, 4, -1)
+
+    assert SingularityFunction(y, 6, 2).diff(y) == 2*SingularityFunction(y, 6, 1)
+    assert SingularityFunction(y, -4, -1).diff(y) == SingularityFunction(y, -4, -2)
+    assert SingularityFunction(y, 4, 0).diff(y) == SingularityFunction(y, 4, -1)
+    assert SingularityFunction(y, 4, 0).diff(y, 2) == SingularityFunction(y, 4, -2)
+
+    assert SingularityFunction(x, -10, 5).diff(evaluate=False) == (
+        Derivative(SingularityFunction(x, -10, 5), x))
+
+    raises(ArgumentIndexError, lambda: SingularityFunction(x, 4, 5).fdiff(2))
+
+
+def test_eval():
+    assert SingularityFunction(x, a, n).func == SingularityFunction
+    assert SingularityFunction(x, 5, n) == SingularityFunction(x, 5, n)
+    assert SingularityFunction(5, 3, 2) == 4
+    assert SingularityFunction(3, 5, 1) == 0
+    assert SingularityFunction(3, 3, 0) == 1
+    assert SingularityFunction(4, 4, -1) == oo
+    assert SingularityFunction(4, 2, -1) == 0
+    assert SingularityFunction(4, 7, -1) == 0
+    assert SingularityFunction(5, 6, -2) == 0
+    assert SingularityFunction(4, 2, -2) == 0
+    assert SingularityFunction(4, 4, -2) == oo
+    assert SingularityFunction(6.1, 4, 5) == 40.84100999999997
```

Comment:
Comparing floats should be done using sympy Float type or you need to use proper floating point comparison.

Nice reference to read: https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html